frontend/app/controllers/spree/products_controller.rb

```diff
@@ -58,5 +58,9 @@ def redirect_if_legacy_path
         redirect_to url_for(params), status: :moved_permanently
       end
     end
+
+    def load_taxonomies
+      defined?(get_taxonomies) ? get_taxonomies : Spree::Taxonomy.includes(root: :children)
```

Comment:
just

```ruby
Spree::Taxonomy.includes(root: :children)
```

and later extension developers can overwrite `load_taxonomies` method